saleor/graphql/giftcard/types.py

```diff
@@ -0,0 +1,45 @@
+from textwrap import dedent
+
+import graphene
+from graphql_jwt.decorators import permission_required
+
+from ...giftcard import models
+from ..core.connection import CountableDjangoObjectType
+
+
+class GiftCard(CountableDjangoObjectType):
+    display_code = graphene.String(
+        description="Code in format with allows displaying in a user interface."
+    )
+    user = graphene.Field(
+        "saleor.graphql.account.types.User",
+        description="The customer who bought a gift card.",
+    )
+
+    class Meta:
+        description = dedent(
+            """
+        A gift card is a prepaid electronic payment card accepted in stores.
+        They can be used during checkout by providing a valid gift
+        card codes. """
+        )
+        only_fields = [
+            "code",
```

Comment:
Question - should full code be visible only to the owner of the voucher? We do expose `displayCode` but still, the `code` itself is also available.

Reply:
I will create custom resolver for `code` and only gift card owner can see it. When we decide to show code only with email we can simply delete it.